code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm

```diff
@@ -108,9 +108,9 @@
 /obj/machinery/atmospherics/components/binary/crystallizer/proc/check_gas_requirements()
 	var/datum/gas_mixture/contents = airs[2]
 	for(var/gas_type in selected_recipe.requirements)
-		if(!contents.gases[gas_type] || !contents.gases[gas_type][MOLES])
-			return FALSE
-	return TRUE
+		if(contents.gases[gas_type] && contents.gases[gas_type][MOLES])
+			return TRUE
+	return FALSE
```

Comment:
The logic here seems to directly contradict what the proc name implies should be going on, what was the actual issue that was causing the behavior you describe?

Reply:
Sometimes the pipes and the machine would clog and no gases would be pulled in, also if you only had one of the gases the machine would stop pulling in even if the gas was the right one.
I could make the proc name singular so that even if there is only one gas the return would be true and the check will pass

Reply:
Or I could name it something like
recipe gas present
Or
check input gas

Reply:
uh now that i think about it, i could remove this check since it would be non necessary now